main.tf

```
@@ -0,0 +1,93 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+
+  backend "azurerm" {
+    resource_group_name  = "Cohort25_JacWoo_ProjectExercise"
+    storage_account_name = "jacwooterraformstate"
+    container_name       = "state"
+    key                  = "terraform.tfstate"
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name     = "Cohort25_JacWoo_ProjectExercise"
+}
+
+resource "azurerm_service_plan" "main" {
+  name                = "${var.prefix}-terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+}
+
+resource "azurerm_linux_web_app" "main" {
+  name                = "${var.prefix}-terraformed-jsw-todo-app"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+
+  site_config {
+    application_stack {
+      docker_image     = "jackiew104/todo-app"
+      docker_image_tag = "prod" # TODO exercise-12: could change this to use prefix if had other images
```

Really minor but you'll see your pipeline mentions this is deprecated; it's a very unexciting move to combine these two together:
```suggestion
      docker_image_name     = "jackiew104/todo-app:prod"
```

[They also suggest](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_web_app#docker_image_name) you specify `docker_registry_url` in the same block (as `https://index.docker.io`) and remove the app_setting entry for the same below (although I believe they're functionally identical!)